Automattic/jetpack
# Social: Add polling flag for share status for better UI

As a part of showing better loading indicator and handle edge cases in post publish panel, we need a flag that indicates that polling for share status is in progress because the current `loading` flag only means that an API request is in flight, it doesn' incorporate the polling interval, i.e. delay in between the API requests.

## Proposed changes:

* Add `polling` flag for share status in social store
* Use polling flag in post publish panel to handle edge cases

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable the share status feature - `define( 'JETPACK_SOCIAL_HAS_SHARE_STATUS', true );`
* Create a post and publish it, sharing to some social media connections
* Confirm that the post publish panel reflects the loading correctly - `Sharing to your social media…`
* Create another post and this time share to Instagram without an image or SIG to simulate failure
* Confirm that the failure notice in the post publish panel works correctly
* To simulate timeout, add `sleep( 65 ); // seconds` before `return` in `Automattic\Jetpack\Publicize\REST_Controller::get_post_share_status()`
* Publish a post again and check the post publish panel again.
* Confirm that you see the message, `"The request to share your post is still in progress."`
* To simulate the edge case, add this before `return`
```php
return rest_ensure_response(
	array(
		'shares' => array(),
		'done'   => true,
	)
);
```
* Publish a post again and check the post publish panel again.
* Confirm that you see the message, `"Your post was not shared."`

<img width="282" alt="image" src="https://github.com/user-attachments/assets/85d375c0-4bce-4bbb-82b0-345f6fe6cf24">
<br>
<img width="278" alt="image" src="https://github.com/user-attachments/assets/490028ca-222e-4222-956c-87cbeb09f006">
